Adding common, bundles, base, and secretstores

schwesig opened the pull request:
fixes https://github.com/nerc-project/operations/issues/280
desiredUpdate: version: 4.14.5 (initial install)
adding basic resources, components, oauth, secrets
https://github.com/nerc-project/operations/issues/354

schwesig committed:
add common, bundles and clusterversion
reducing kustomization, setting clusterversion to currently rolled out one
finalising basic config
linter fixes
adding secretstores
Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

computate reviewed (approved):
Looks good, you might just need to ask @larsks about the GitHub Auth Client ID.

larsks commented on cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -0,0 +1,73 @@
+resources:
+- ../common
+- ../../bundles/node-feature-discovery
+- ../../bundles/patch-operator
```
Do we need the patch operator? We've tried to avoid it as much as possible.

schwesig replied:
you are right. We should not need it.

larsks commented on cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -0,0 +1,73 @@
+- ../common
+- ../../bundles/node-feature-discovery
+- ../../bundles/patch-operator
+# - ../../bundles/rhods-operator
```
Please do not leave commented-out items in these files. If we need them we'll add them in a future pull request. I'm pretty sure we have no plans to run RHODS on the observability cluster.

schwesig replied:
good point. agreed.

larsks commented on cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -0,0 +1,73 @@
+      - name: github
+        github:
+#          clientID: $(GITHUB_CLIENT_ID)
+#          clientSecret: $(GITHUB_CLIENT_SECRET)
```
This shouldn't need to be patched; it's always called "github-client-secret".

schwesig replied:
good to know, I was following the example from some other overlays here.

larsks commented on cluster-scope/overlays/nerc-ocp-obs/secretstores/kustomization.yaml:
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- acct-mgt
```
You don't need this (we're not running the account management tools on the obs cluster).

schwesig replied:
i wasn't sure, if needed for different right settings than the infra (outside vpn).
following your input here.

larsks commented on cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -0,0 +1,73 @@
+      identityProviders:
+      - name: github
+        github:
+#          clientID: $(GITHUB_CLIENT_ID)
```
This should be the client id from the `nerc-ocp-obs` application found at <https://github.com/organizations/OCP-on-NERC/settings/applications>. Are you able to view that?

schwesig replied:
@larsks I am not able to view that :-(

schwesig replied:
received, thanks

schwesig committed:
removed patch operator, comments for future issue,
added github client-id and external client-secret,
triggered by comments from reviews
clean up for final commit
added externalsecrets
adding cert and aws route 54

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

schwesig committed:
Merge remote-tracking branch 'origin/main' into 202312_advancedConfigNewObsCluster
